src/Core/Command.rei

```diff
@@ -0,0 +1,28 @@
+[@deriving show]
+type t('msg) = {
+  id: string,
+  title: option(string),
+  category: option(string),
+  icon: option([@opaque] IconTheme.IconDefinition.t),
+  isEnabledWhen: WhenExpr.t,
+  msg: [ | `Arg0('msg) | `Arg1(Json.t => 'msg)],
```

Comment:
This polymorphic variant exists to support commands with contextual arguments. Perhaps not the most elegant approach, but it works well. An alternative might be to distinguish more strongly between extension-sourced commands and built-in commands.

Reply:
Interesting!